internal/queue/consumer.go

```diff
@@ -0,0 +1,117 @@
+// Package queue implements the message queue functionality using RabbitMQ.
+package queue
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Konstantsiy/image-converter/internal/config"
+	"github.com/Konstantsiy/image-converter/internal/converter"
+	"github.com/Konstantsiy/image-converter/internal/repository"
+	"github.com/Konstantsiy/image-converter/internal/storage"
+	"github.com/Konstantsiy/image-converter/pkg/logger"
+	"github.com/streadway/amqp"
+)
+
+// Consumer listens to the queue and processes outgoing messages.
+type Consumer struct {
+	client  *rabbitMQClient
+	repo    *repository.Repository
+	storage *storage.Storage
+}
+
+func NewConsumer(repo *repository.Repository, storage *storage.Storage, conf *config.Config) (*Consumer, error) {
+	client, err := initRabbitMQClient(conf)
+	if err != nil {
+		return nil, err
+	}
+	return &Consumer{repo: repo, storage: storage, client: client}, nil
+}
+
+// ListenToQueue listens to the queue channel in a separate goroutine.
+func (c *Consumer) ListenToQueue() error {
+	err := c.client.ch.Qos(1, 0, false)
+	if err != nil {
+		return fmt.Errorf("can't configure QoS: %w", err)
+	}
```

Comment:
I'd move that to `initRabbitMQClient`

Reply:
> I'd move that to `initRabbitMQClient`

But we need the `Qos` method exclusively for consuming queue messages. There are only general functions in `initRabbitMQClient`.

Reply:
ok then